src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.Tests.cs

```diff
@@ -79,6 +79,28 @@ public void Constructor_NoEntryInsertion_WritesNothing()
             Assert.Equal(0, archiveStream.Length);
         }
 
+        [Fact]
+        public void Write_To_UnseekableStream()
+        {
+            MemoryStream stream = new MemoryStream();
+            using WrappedStream wrapped = new WrappedStream(stream, canRead: true, canWrite: true, canSeek: false);
+
+            using (TarWriter writer = new TarWriter(wrapped, leaveOpen: true))
+            {
+                PaxTarEntry paxEntry = new PaxTarEntry(TarEntryType.RegularFile, "file.txt");
+                writer.WriteEntry(paxEntry);
+            } // The final records should get written, and the length should not be set because position cannot be read
+
+            stream.Seek(0, SeekOrigin.Begin); // Rewind the base stream (wrapped cannot be rewinded)
+
+            using TarReader reader = new TarReader(wrapped);
```

Comment:
I think it's a little clearer to use a using block to match the writer here.

Reply:
Sure, I can wrap this.